pymeasure/instruments/tektronix/tek371A.py

```diff
@@ -0,0 +1,859 @@
+                      f'{self.x_scale_factor} {self.horizontal_units} \n' \
+                      f'Number Horizontal Range: ' \
+                      f'{self.horizontal_range} \n' \
+                      f'Number Horizontal Resolution Points: ' \
+                      f'{self.n_horizontal_resolution_points}\n' \
+                      f'Horizontal Offset: ' \
+                      f'{self.horizontal_offset} \n' \
+                      f'Horizontal Units: ' \
+                      f'{self.horizontal_units} \n' \
+                      f'Vertical Y Scale Factor: ' \
+                      f'{self.y_scale_factor} {self.vertical_units} \n' \
+                      f'Number Vertical Range: ' \
+                      f'{self.vertical_range} \n' \
+                      f'Number Vertical Resolution Points: ' \
+                      f'{self.n_vertical_resolution_points} \n' \
+                      f'Vertical Offset: ' \
+                      f'{self.vertical_offset} \n' \
+                      f'Vertical Units: ' \
+                      f'{self.vertical_units} \n' \
+                      f'Step Size: ' \
+                      f'{self.step_size} \n' \
+                      f'Step Offset: ' \
+                      f'{self.step_offset} \n\n'
+
+        return description
+
+
+class Curve:
+    """ Represents the Tektronix Curve Data from the Tektronix model 371A
+    and provides a high-level interface for interacting with and translate
+    data to electrical magnitudes.
+    """
+
+    # Curve data sets are usually much longer than any other kind.
+
+    # The major part of a curve is a sequence of binary coded numbers,
+    # which is prefixed by a 25 character ASCII string identifying the curve.
+
+    # Typically a set of curve data will be about 4122 bytes long, with most of the bytes being
+    # binary-coded numbers. Thus, most of the string of data is not directly readable, but must
+    # be interpreted by the controller. An example might look like this. CURVE CURVID:”INDEX 9”,
+    # %NNXXYYXXYY . . . XXYYC This example breaks down as follows. It starts with an ASCII string
+    # of 25 characters: CURVE CURVID:”INDEX 9”,% This is followed by a series of binary bytes.
+    # The first of these is two bytes giving the number of data bytes to follow, plus one (
+    # typically 4097): NN Then come the 4096 data bytes. Each of the 1024 data points on the
+    # curve is represented by four bytes, 2 for the 10 bits of the X coordinate and 2 for the 10
+    # bits of the Y coordinate: XXYYXXYY . . . XXYY And finally there is one byte which is the
+    # checksum for the preceding 4098 data bytes.
+
+    # The first XXYY of the 1024 points is the first point taken whe we execute the sweep.
+    # Last point should be the coordinates 0,0.
+    # XXYY is coordinate point. Thus, we have to translate it for getting electrical values.
+
+    def __init__(self,
+                 waveform_preamble: WaveformPreamble,
```

Comment:
In pymeasure, we do not use type hinting in this form.
Instead, we add define the parameters in the (class) docstring, where it is possible to state the type as well:
~~~
:param waveform_preample: Some description.
:type waveform_preample: WaveformPreamble
~~~

Reply:
got it. Many thanks